### ISSUE TITLE
`zig c++ -emit-llvm` error

### ISSUE THREAD

**Issue by lolgab** [ziglang/zig#6425]
Got an error trying to emit-llvm with `zig c++`:
```bash
./zig c++ -emit-llvm ~/test.cpp
Build Dependencies... c++ [2] Compile C Objects [37/37] vector.cpp... Cannot open zig-cache/o/Cw6OiveYOXWyQvD-DwLoiWkkFZJv2JkU4YBLyE9TDsqW380Ykye6jaX_UDvCYTs1/test.o: Exec format error
```
I'm using Zig version: `zig-macos-x86_64-0.6.0+a50260470`. Same thing happens with `0.6.0`.

This is the simple hello world I'm testing:
```c++
#include <iostream>
using namespace std;

int main()
{
  cout << "Hello world.\n";
  return 0;
}
```
## Other options
```bash
./zig c++ -emit-llvm -S ~/test.cpp
```
Works but it prints the output to the `stdout` instead of generating a file ( like `clang` does ).
Clang supports to write to a specific output with:

```bash
clang++ -emit-llvm -S ~/test.cpp -o outputfile.ll
```
This is not supported too.

**Comment by lolgab** [comment]
Thank you @andrewrk!

**Comment by x4e** [comment]
I got a similar error after updating zig from 0.8.0 to 0.8.1 (which I believe includes your fix, d11f42c). 
It only happens when cross-compiling a c file, with the argument "-flto=full", (from Linux x86_64) to MacOS x86_64.

```
warning(link): unexpected LLD stderr:
Cannot open zig-cache/o/93a2874c6384d372d7972f27dfd13349/main.o: Exec format error
```

**Comment by andrewrk** [comment]
> which I believe includes your fix, [d11f42c](https://github.com/ziglang/zig/commit/d11f42c2b24e80388e1ea648ee97fae612fd76a4)).

8.0.1 does not include the fix unfortunately.

**Comment by x4e** [comment]
Ok, I will test again when that commit gets included in a release. Thanks!
